sunkit_image/granule.py

```diff
@@ -0,0 +1,298 @@
+"""
+This module contains functions that will segment images for granule detection.
+"""
+import logging
+
+import numpy as np
+import scipy.ndimage as sndi
+import skimage
+from sklearn.cluster import KMeans
+
+import sunpy
+import sunpy.map
+
+__all__ = [
+    "segment",
+    "get_threshold",
+    "trim_intergranules",
+    "mark_brightpoint",
+    "kmeans_segment",
+    "correlation",

```

Comment:
That makes a lot of sense. Just to clarify, are you suggesting getting rid of the `correlation` function (and also `segment_kmeans` I presume?) or just not naming it at the top of the file?

Reply:
Based on the subsequent discussion and clarification from @nabobalis, it sounds like `correlation` and `segment_kmeans` can be removed altogether. For the other functions, I would suggest (a) removing them from the `__all__` list and also prepending the function name with a `_` to make it not part of the public API (e.g. `get_threshold` becomes `_get_threshold`.

Reply:
Ok, I will just get rid of them. Although from the comment below, @kreardon does seem to prefer keeping them. @kreardon are you ok with them going?

Reply:
I describe some good use cases in may reply to the comment on line 294. I think keeping `correlation` (but with my suggestion to rename it to `overlap` or `segment_overlap`, in part to avoid the namespace confusion Will mentions) as a user callable function would help in a variety of user workflows.

Reply:
Ok, so sounds like the best option is to keep them following option 2 I outlined below, except to keep `segment_kmeans` seperate. @wtbarnes does that sound ok to you?

Reply:
That sounds good to me!

Reply:
Sorry, one last minor concern on this thread. It looks like the only thing that depends on scikit-learn is this semi-peripheral `segment_kmeans` function. scikit-learn, while a commonly-used tool, is not a light dependency and if it is not part of the primary functionality in segment, it doesn't seem worth the cost of picking up a dependency just to keep this around.

So I guess my (actually!) final preference would be to remove the kmeans function and keep `correlation` with the suggested renaming mentioned by Kevin.

Reply:
Ok! I can leave in just `correlation` (perhaps rename to `compute_ovelap` or something), and just recomend that the second map is generated using something like kmeans segment.